docs/Contributing/high-level-architecture.md

```diff
@@ -0,0 +1,192 @@
+# High level architecture
+
+- [Overview](#overview)
+- [Main System Components](#main-system-components)
+
+## Overview
+
+Add text
+
+## Main System Components
+
+```mermaid
+graph LR;
+    
+    subgraph Development
+        fleet_release_owner[Fleet Release<br>Owner];
+    end
+
+    subgraph Agent
+        orbit[orbit];
+        desktop[Fleet Desktop<br>Tray App];
+        osqueryd[osqueryd];
+        desktop_browser[Fleet Desktop<br> from Browser];
+    end
+
+    subgraph Customer Cloud
+        fleet_server[Fleet<br>Server];
+        db[DB];
+        redis[Redis<br>Live queries' results <br>go here];
+        prometheus[Prometheus Server];
+    end
+
+    subgraph FleetDM Cloud
+        tuf["<a href=https://theupdateframework.io/>TUF</a> file server<br>(default: <a href=https://tuf.fleetctl.com>tuf.fleetctl.com</a>)"];
+        datadog[DataDog metrics]
+        heroku[Usage Analytics<br>Heroku]
+        log[Optional Log Location<br>Store logs here]
+    end
+
+    subgraph Customer Admin
+        frontend[frontend code]
+    end
+
+
+    fleet_release_owner -- "Release Process" --> tuf;
+
+    orbit -- "Fleet Orbit API (TLS)" --> fleet_server;
+    orbit -- "Auto Update (TLS)" --> tuf;
+    desktop -- "Fleet Desktop API (TLS)" --> fleet_server;
+    osqueryd -- "osquery<br>remote API (TLS)" --> fleet_server;
+    desktop_browser -- "My Device API (TLS)" --> fleet_server;
+
+    heroku -- "Metrics from all customers" --> datadog;
+
+    fleet_server <== "Read/Write" ==> db;
+    fleet_server <== "Read/Write" ==> redis;
+    redis <==> db;
+
+    prometheus ==> fleet_server;
+    fleet_server -- "metrics" --> heroku;
+    fleet_server -- "queries results" --> log;
+
+    frontend <== "API" ==> fleet_server;
+
+```
+
+
+
+## The path of Live Query
+
+### 1 - IT person initiates the query
+```mermaid
+graph LR;
+    it_person[IT person<br>Starts a live query];
```

Comment:
@zhumo do we refer to customers who log into the Fleet console as "IT Person"?  Just want to make sure we're being consistent with our customer personas

Reply:
Let's go with "Fleet user" to be more generic. IT not the only job function that uses Fleet. If this is not for all Fleet users, but the admins of Fleet, then we would say "Fleet admin"